pkg/bloomgateway/bloomgateway.go

```diff
@@ -339,67 +341,45 @@ func (g *Gateway) FilterChunkRefs(ctx context.Context, req *logproto.FilterChunk
 
 	tasks := make([]Task, 0, len(seriesWithBloomsPerDay))
 	for _, seriesWithBounds := range seriesWithBloomsPerDay {
-		task, err := NewTask(tenantID, seriesWithBounds, req.Filters)
+		task, err := NewTask(ctx, tenantID, seriesWithBounds, req.Filters)
 		if err != nil {
 			return nil, err
 		}
 		tasks = append(tasks, task)
-		expectedResponses += len(seriesWithBounds.series)
+		numSeries += len(seriesWithBounds.series)
 	}
 
 	g.activeUsers.UpdateUserTimestamp(tenantID, time.Now())
 
-	errCh := make(chan error, 1)
-	resCh := make(chan v1.Output, 1)
+	tasksCh := make(chan Task)
```

Comment:
we know tasks count upfront
```suggestion
	tasksCh := make(chan Task, len(tasks))
```

Reply:
I explicitly want an unbuffered channel.